server/routes/posts.js

```diff
@@ -38,30 +44,107 @@ module.exports = (() => {
       })
     })
 
-    router.post('/create', (req, res) => {
-      
+
+
+    /* Endpoint to provide partial list of posts based on keyword search */
+    router.get('/list/:keyword', (req,res) => {
+      console.log("Endpoint: Listing Posts Based on Keyword Match ")
+     Post.find({body: {$regex: req.params.keyword}}, function(err, posts) {
+        if (err) throw err;
+        if (!posts) {
+          const init = new Post({
+            id: 0,
+            username: 'ragingbull',
+            displayName: 'Raging Bull',
+            date: Date.now(),
+            title: 'Greetings from California State University, ' +
+              'Dominguez Hills in Carson, CA!',
+            body: 'This post serves as a placeholder.'
+          })
+
+          init.save(err => {
+            if (err) throw err
+            console.log('Sample post initialized.')
+            res.json({ posts: init })
+          })
+        }
+        else {
+          console.log('Posts retrieved: ', posts)
+          res.json({ posts })
+        }        
+      })
+    });//End Endpoint
+
+
+
+
+    /* Endpoint: Delete Single Post Based on  */
+    /* new attempt */
+    router.get('/delete/:id', (req, res) => {
```

Comment:
delete has to be a POST request

Reply:
I think John said single delete is ok with GET, but to delete multiple posts, I will need POST

Reply:
Ok but deletes are clean with post. But if professor said so, then fine